lib/custom_html_renderer.rb

```diff
@@ -0,0 +1,27 @@
+require 'middleman-core/renderers/redcarpet'
+
+class CustomHTMLRenderer < Middleman::Renderers::MiddlemanRedcarpetHTML
+  include Redcarpet::Render::SmartyPants
+
+  # This is effectively a reimplementation of
+  # https://github.com/vmg/redcarpet/blob/master/ext/redcarpet/html.c#L307-L325
+  # in Ruby.
+  def header(text, level)
+    fragment = header_anchor(text)
+    <<~EOS
```

Comment:
It's worth noting that the "squiggly heredoc" syntax is only available in Ruby 2.3+ – I'm assuming that's not a problem.